src/dmtest/vdo/dedupe_tests.py

```diff
@@ -0,0 +1,87 @@
+from dmtest.assertions import assert_raises, assert_equal, assert_near
+from dmtest.vdo.utils import standard_stack, standard_vdo
+import dmtest.fs as fs
+import dmtest.process as process
+import dmtest.units as units
+import dmtest.vdo.stats as stats
+import dmtest.test_register as reg
+
+import os
+import re
+import threading
+import logging as log
+import time
+
+#---------------------------------
+
+def make_delta_stats(stats_post, stats_pre):
+    """Recursive function that transforms only these dictionary values which keys correspond to a `mapping` dict."""
+    if isinstance(stats_post, dict):
+        node = {}
+        for key, value in stats_post.items():
+            node[key] = make_delta_stats(value, stats_pre[key])
+        return node
+    elif isinstance(stats_post, int):
+        return stats_post - stats_pre
+
+def get_dataset_config():
+    config = {
+        "gen.large.num" : 32,
+        "gen.large.min" : 4 * 1024 * 1024,
+        "gen.large.max" : 4 * 1024 * 1024,
+    }
+    return config
+
+def verify_dedupe(vdo, fs_type, dedupe):
+    fs = fs_type(vdo)
+    fs.format(discard=True)
+
+    with fs.mount_and_chdir("./mnt", discard=False):
+        stats_pre = stats.vdo_stats(vdo)
+        config = get_dataset_config()
+        config["gen.large.numCoalescent"] = int((config["gen.large.num"] * dedupe) / 100)
+        config["gen.root.dir"] = os.getcwd()
+        options = "--stdout --logDir=../"
+        for k, v in config.items():
+            options += " -D" + k + "=" + str(v)
+        process.run("../src/scripts/gen_dataset.py " + options)
+        for subdir, dirs, files in os.walk(os.getcwd()):
+            for file in files:
+                path = os.path.join(subdir, file)
+                with open(path, 'r') as out:
+                    out.flush()
+                    os.fsync(out.fileno())
```

Comment:
I think running "sync -f {path}" should do the equivalent of this whole loop. I don't know if the syncfs() system call is trivially available from Python.

Reply:
I'll try doing this

Reply:
Done

Reply:
You changed another use of fsync but not this tree-walking one

Reply:
fixed